modules/gapi/samples/privacy_masking_camera.cpp

```diff
@@ -0,0 +1,215 @@
+        for (int i = 0; i < MAX_PROPOSALS; i++) {
+            const float image_id   = data[i * OBJECT_SIZE + 0];
+            const float label      = data[i * OBJECT_SIZE + 1];
+            const float confidence = data[i * OBJECT_SIZE + 2];
+            const float rc_left    = data[i * OBJECT_SIZE + 3];
+            const float rc_top     = data[i * OBJECT_SIZE + 4];
+            const float rc_right   = data[i * OBJECT_SIZE + 5];
+            const float rc_bottom  = data[i * OBJECT_SIZE + 6];
+
+            if (image_id < 0.f) {
+                break;    // marks end-of-detections
+            }
+            if (confidence < 0.5f) {
+                continue; // skip objects with low confidence
+            }
+            if (filter_label != -1 && static_cast<int>(label) != filter_label) {
+                continue; // filter out object classes if filter is specified
+            }
+
+            cv::Rect rc;  // map relative coordinates to the original image scale
+            rc.x      = static_cast<int>(rc_left   * upscale.width);
+            rc.y      = static_cast<int>(rc_top    * upscale.height);
+            rc.width  = static_cast<int>(rc_right  * upscale.width)  - rc.x;
+            rc.height = static_cast<int>(rc_bottom * upscale.height) - rc.y;
+            out_objects.emplace_back(rc & surface);
+        }
+    }
+};
+
+GAPI_OCV_KERNEL(OCVToMosaic, ToMosaic) {
+    static void run(const std::vector<cv::Rect> &in_plate_rcs,
+                    const std::vector<cv::Rect> &in_face_rcs,
+                          std::vector<cv::gapi::wip::draw::Prim> &out_prims) {
+        out_prims.clear();
+        const int BLOCK_SIZE = 24;
+        const auto cvt = [](cv::Rect rc) {
+            // Align the mosaic region to mosaic block size
+            const int dw = BLOCK_SIZE - (rc.width  % BLOCK_SIZE);
+            const int dh = BLOCK_SIZE - (rc.height % BLOCK_SIZE);
+            rc.width  += dw;
+            rc.height += dh;
+            rc.x      -= dw / 2;
+            rc.y      -= dh / 2;
+            return cv::gapi::wip::draw::Mosaic{rc, 24, 0};
+        };
+        for (auto &&rc : in_plate_rcs) { out_prims.emplace_back(cvt(rc)); }
+        for (auto &&rc : in_face_rcs)  { out_prims.emplace_back(cvt(rc)); }
+    }
+};
+
+} // namespace custom
+
+int main(int argc, char *argv[])
+{
+    cv::CommandLineParser cmd(argc, argv, keys);
+    cmd.about(about);
+    if (cmd.has("help")) {
+        cmd.printMessage();
+        return 0;
+    }
+    const std::string input = cmd.get<std::string>("input");
+    const bool no_show = cmd.get<bool>("noshow");
+    const bool run_trad = cmd.get<bool>("trad");
+
+    cv::GMat in;
+    cv::GMat blob_plates = cv::gapi::infer<custom::VehLicDetector>(in);
+    cv::GMat blob_faces  = cv::gapi::infer<custom::FaceDetector>(in);
+    cv::GArray<cv::Rect> rc_plates = custom::ParseSSD::on(blob_plates, in, 2);
+    cv::GArray<cv::Rect> rc_faces  = custom::ParseSSD::on(blob_faces, in, -1);
```

Comment:
> 2
> -1

Consider extracting of magic numbers and give names for them:
```
const int ssd_label_plate = 2;
const int ssd_label_face = -1;
```

Reply:
-1 is not a face label BTW, it assumes no filtering is required (a single-class detector).
Maybe an inline comment would be enough?